text/0000-landmarks.md

```diff
@@ -0,0 +1,43 @@
+# Landmark observations
+Add landmarks support to Cartographer.
+
+## Summary
+[summary]: #summary
+Cartographer will be able to work with a new type of sensor data that contains IDs and locations of landmarks.
+
+## Motivation
+[motivation]: #motivation
+Landmarks translation/orientation can improve SLAM quality. Landmarks rendered on the map can be useful for inspection. 
+
+## Approach
+[approach]: #approach
+Landmarks are sparse features of geometry that we want to map. Unlike dense range data, landmarks are assumed to be sparse in comparison. In order to support using them, current SLAM approach has to be generalized. 
+
+### Sensor
+Each landmark will have an ID and a position relative to the robot with or without orientation. A landmark observation can contain several landmarks at a time.
+
+### Submap
+Submaps for 2D and 3D geometries are occupancy grids that contain accumulated range data.  There can be two approaches to add landmarks to the submaps:
+
+1. Submap holds a dictionary with landmark data keyed by landmark IDs;
+2. Every cell of the grid is modified to hold various features, for example, likelihood of landmarks, ML features. 
```

Comment:
So to rephrase that: 1. is something like a maximum posterior approach, while with 2. you want to model the distribution. How would you use this distribution knowledge to approve mapping? How would e.g. the ContraintBuilder and PoseGraph use this information?

Reply:
What is an  "ML feature" in this context?

Reply:
To Michael: any ML features, for example, if robot can recognize bounding boxes of different objects while building a map, etc. Or features for visual SLAM.

Reply:
To Christoph: the approach is the same, it is more about how we store landmarks in Submap class: do we store them separately from the probability grid or not.

Reply:
Oh are you saying that in the dictionary approach you would store for a given landmark different possible poses along with a probability also?